src/content/advanced/gitops/environments/index.md

```diff
@@ -0,0 +1,386 @@
+          namespace: org-${organization}
+          priority: 110
+    target:
+      group: application.giantswarm.io
+      kind: App
+      name: ${cluster_name}
+      namespace: org-\${organization}
+  - patch: |
+      - type: merge
+        op: add
+        path: /spec/extraConfigs/0
+        value:
+          name: ${cluster_name}-default-apps-dev-config
+          namespace: org-${organization}
+          priority: 110
+    target:
+      group: application.giantswarm.io
+      kind: App
+      name: ${cluster_name}-defaul-apps
+      namespace: org-\${organization}
+resources:
+  - ../../../../../../../../bases/clusters/capa/template/
+```
+
+In addition, we are creating a configuration file `cluster_config.yaml` with the values we are changing from the cluster base:
+
+```yaml
+controlPlane:
+  replicas: 1
+machinePools:
+- instanceType: m5.large
+  maxSize: 5
+  minSize: 3
+  name: machine-dev-pool0
+  rootVolumeSizeGB: 100
+network:
+  availabilityZoneUsageLimit: 1
+```
+
+We want to set a specific version for some default apps installed in the cluster too, so we create a file `default_apps_config.yaml` with the specific values.
+
+```yaml
+userConfig:
+  apps:
+    cilium:
+      version: 0.6.1
+    coreDNS:
+      version: 1.13.0
+```
+
+The final structure of this folder would be:
+
+```nohighlight
+bases/environments/stages/dev
+└── example_cluster
+    ├── cluster_config.yaml
+    ├── default_apps_config.yaml
+    └── kustomization.yaml
+```
+
+And pretty much that is it for the development cluster. Let's look at the production cluster next, created with minor differences.
+
+#### The production cluster
+
+Let's create the `Kustomization` for the production cluster environment template.
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+buildMetadata: [originAnnotations]
+configMapGenerator:
+  - files:
+    - values=cluster_config.yaml
+    name: ${cluster_name}-prod-config
+    namespace: org-${organization}
+  - files:
+    - values=default_apps_config.yaml
+    name: ${cluster_name}-default-apps-prod-config
+    namespace: org-${organization}
+generatorOptions:
+  disableNameSuffixHash: true
+kind: Kustomization
+patches:
+  - patch: |
+      - type: merge
+        op: add
+        path: /spec/extraConfigs/0
+        value:
+          name: ${cluster_name}-prod-config
+          namespace: org-${organization}
+          priority: 110
+    target:
+      group: application.giantswarm.io
+      kind: App
+      name: ${cluster_name}
+      namespace: org-\${organization}
+  - patch: |
+      - type: merge
+        op: add
+        path: /spec/extraConfigs/0
+        value:
+          name: ${cluster_name}-default-apps-prod-config
+          namespace: org-${organization}
+          priority: 110
+    target:
+      group: application.giantswarm.io
+      kind: App
+      name: ${cluster_name}-defaul-apps
+      namespace: org-\${organization}
+resources:
+  - ../../../../../../../../bases/clusters/capa/v0.21.0/
```

Comment:
```suggestion
  - ../../../../../../../../bases/clusters/capa
```
I would like to preserve same template and not make it complex 🙏

Reply:
this is not correct with versioning. You need to specify "capa/template" or "capa/v0xxx" in order to use one of them.

Reply:
yeah then I would vote for just template and not overcomplicate in this example

Reply:
BTW fixed